pkgs/by-name/ni/nixos-render-docs/src/nixos_render_docs/md.py

```diff
@@ -537,22 +542,37 @@ def _block_titles(block: str) -> Callable[[markdown_it.MarkdownIt], None]:
     non-title heading since those would make toc generation extremely complicated.
     """
     def block_titles(state: markdown_it.rules_core.StateCore) -> None:
-        in_example = [False]
+        in_example = [None]
         for i, token in enumerate(state.tokens):
             if token.type == open:
                 if state.tokens[i + 1].type == 'heading_open':
                     assert state.tokens[i + 3].type == 'heading_close'
                     state.tokens[i + 1].type = title_open
                     state.tokens[i + 3].type = title_close
                 else:
-                    assert token.map
-                    raise RuntimeError(f"found {block} without title in line {token.map[0] + 1}")
-                in_example.append(True)
+                    raise SrcError(
+                        src=state.src,
+                        description=f"found {block} without title",
+                        token=token,
+                    )
+                in_example.append(token)
```

Comment:
By tracking the whole block token, we can show the user where the parser thinks the block starts and stops, making it a lot clearer when they're missing a delimiter.